catalyst-gateway/bin/src/service/api/documents/post_document_index_query/query_filter.rs

```diff
@@ -181,16 +181,12 @@ impl TryFrom<DocumentIndexQueryFilter> for DocsQueryFilter {
         if let Some(reply) = value.reply {
             db_filter = db_filter.with_reply(reply.try_into()?);
         }
-        if let Some(brand) = value.brand {
-            db_filter = db_filter.with_brand_id(brand.try_into()?);
+        for param in [value.brand, value.campaign, value.category]
```

Comment:
I think it could be just, without `.into_iter().flatten()`. Why it is needed to have `flatten` at this place ?
```
for param in [value.brand, value.campaign, value.category]
 ...
 ```